blockchain/src/chain_store.rs

```diff
@@ -95,8 +205,12 @@ impl ChainStore {
         };
 
         if include_body {
-            if let Some(block) = txn.get(&self.block_table, hash) {
-                chain_info.head = block;
+            // Check the stored block table, then the pushed block table
```

Comment:
We could eliminate one database fetch operation here by storing a `pushed` flag in `ChainInfo`, which tells us which block table to use.

Reply:
Just implemented this but used the already present `chain_info.on_main_chain` flag.